src/api/coderApi.ts

```diff
@@ -198,68 +223,62 @@ export class CoderApi extends Api {
 				throw new Error("No base URL set on REST client");
 			}
 
-			const baseUrl = new URL(baseUrlRaw);
-			const token = this.getAxiosInstance().defaults.headers.common[
-				coderSessionTokenHeader
-			] as string | undefined;
+			return this.createOneWayWebSocket<TData>(socketConfigs);
+		};
 
-			const headersFromCommand = await getHeaders(
-				baseUrlRaw,
-				getHeaderCommand(vscode.workspace.getConfiguration()),
-				this.output,
-			);
+		if (enableRetry) {
+			return this.createReconnectingSocket(socketFactory, configs.apiRoute);
+		}
+		return socketFactory();
+	}
 
-			const httpAgent = await createHttpAgent(
-				vscode.workspace.getConfiguration(),
-			);
+	private async createOneWayWebSocket<TData>(
```

Comment:
The existence of `createOneWayWebSocket` to me implies that `createWebSocket` is two-way socket, but it actually also creates a one-way web socket, except potentially reconnecting.  Maybe we can have just `createOneWayWebSocket` and it handles `enableRetry`?

Or, we could rename `createWebSocket` to `createReconnectingOneWayWebSocket` or something like that.